pramen/core/src/main/scala/za/co/absa/pramen/core/runner/task/TaskRunnerMultithreaded.scala

```diff
@@ -38,22 +39,57 @@ class TaskRunnerMultithreaded(conf: Config,
                               journal: Journal,
                               pipelineState: PipelineState,
                               runtimeConfig: RuntimeConfig) extends TaskRunnerBase(conf, bookkeeper, journal, runtimeConfig, pipelineState) {
+  private val log = LoggerFactory.getLogger(this.getClass)
+
   private val executor: ExecutorService = newFixedThreadPool(runtimeConfig.parallelTasks)
   implicit private val executionContext: ExecutionContextExecutorService = fromExecutorService(executor)
 
+  private val maxResources = runtimeConfig.parallelTasks
+  private val availableResources: Semaphore = new Semaphore(maxResources)
```

Comment:
I'm thinking making the semaphore 'fair'. Otherwise, it could happen that easy jobs can block hard jobs for some time.

```suggestion
  private val availableResources: Semaphore = new Semaphore(maxResources, true)
```